src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/HeaderFileBuilder.java

```diff
@@ -105,13 +105,17 @@ void addStaticFunctionWrapper(String javaName, String nativeName, MethodType mty
         if (!mtype.returnType().equals(void.class)) {
             builder.append("return (" + mtype.returnType().getName() + ")");
         }
-        builder.append(methodHandleGetCallString(javaName, nativeName, mtype, desc, varargs) + ".invokeExact(" + String.join(", ", pExprs) + ");\n");
+        builder.append("Objects.requireNonNull(");
+        builder.append(methodHandleGetCallString(javaName, nativeName, mtype, desc, varargs));
+        builder.append(", \"no such native function: ");
```

Comment:
Nice! I wonder if here we should use "no such" vs. some other form, e.g. "unresolved". I also wonder if NPE is the right exception here - maybe a NoSuchMethodException, or LinkageError might be better?